### PR TITLE
[READY] Refactors wet floors to a component

### PR DESCRIPTION
**Pull request by silicons**
Tin.
:cl:
rscadd: Wet floors now get their duration decreased by 0.1 seconds per degree above 0C the air temperature on it is, and instantly dry above 100C. Below 0C they will freeze into ice and will not automatically dry.
rscadd: Floors can now have different types of wet at the same time! Wet strength/type is now a bitflag.
/:cl:

### CODE REVIEW

**Comment by arsserpentarium** [comment]
can floors be wet with different reagents?

**Comment by kevinz000** [comment]
@arsserpentarium yes

**Comment by ShizCalev** [comment]
conflict.

File: code/__DEFINES/components.dm [hunk 1]
```diff
@@ -86,3 +86,8 @@
 
 #define CALTROP_BYPASS_SHOES 1
 #define CALTROP_IGNORE_WALKERS 2
+
+//Component Specific Signals
+//Wet floors
+#define COMSIG_TURF_IS_WET "check_turf_wet"							//(): Returns bitflags of wet values.
+#define COMSIG_TURF_MAKE_DRY "make_turf_try"						//(max_strength, immediate, duration_decrease = INFINITY): Returns bool.
```
Comment (Poojawa) [inline code/__DEFINES/components.dm]: make_turf_try

try to by dry, baby.

**Pull request merged**